### DISCUSSION TITLE
How to set node affinity on build pods for KafkaConnect

### DISCUSSION THREAD

**Discussion by jslusher** [strimzi/strimzi-kafka-operator#10758] [Q&A]
Is there a way to set node affinity on KafkaConnect build pods?

I ran into an issue where the build pod ran on `arm64` but I have node affinity set on the connect cluster itself for `amd64`. Is there a way to guarantee the build pod runs on the same arch as the connect pods?

Also, is there a way to initiate a manual build? If this is something I can't guarantee I want to at least be able to rebuild the pod until it builds on `amd64`.

I get this on the pod when it tries to start:

```
exec /opt/kafka/kafka_connect_run.sh: exec format error
```

**Comment by scholzj** [comment]
You should be able to set it int he `buildPod` template? In `.spec.template.buildPod...`: https://strimzi.io/docs/operators/latest/full/configuring.html#type-PodTemplate-reference
